crates/ethereum-rpc/src/gas_price/cache.rs

```diff
@@ -8,8 +8,10 @@ use sov_modules_api::WorkingSet;
 
 /// Cache for gas oracle
 pub struct BlockCache<C: sov_modules_api::Context> {
-    // Assuming number_to_hash and cache are always in sync
+    /// Assuming number_to_hash and cache are always in sync
+    /// Acquire after `cache`
     number_to_hash: Mutex<LruMap<u64, B256, ByLength>>, // Number -> hash mapping
```

Comment:
Because it's a (A or B) -> C it's better to keep two lru in 1 struct behind 1 mutex.

Reply:
Opening an issue for this